lib/rules/no-static-typos.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * @fileoverview Prevent casing typos when declaring propTypes, contextTypes and defaultProps
+ */
+'use strict';
+
+var Components = require('../util/Components');
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+var STATIC_CLASS_PROPERTIES = ['propTypes', 'contextTypes', 'childContextTypes', 'defaultProps'];
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Prevent casing typos when declaring static class properties',
+      category: 'Stylistic Issues',
+      recommended: false
+    },
+    schema: []
+  },
+
+  create: Components.detect(function(context, components, utils) {
+    function reportErrorIfCasingTypo(node, propertyName) {
+      STATIC_CLASS_PROPERTIES.forEach(function(CLASS_PROP) {
+        if (CLASS_PROP.toLowerCase() === propertyName.toLowerCase() && CLASS_PROP !== propertyName) {
+          context.report({
+            node: node,
+            message: 'Typo in static class property declaration'
```

Comment:
It seems like this might be pretty straightforward to write a fixer for. Would you be interested in adding that?